dashboard/db/migrate/20210609212022_add_published_state_to_script.rb

```diff
@@ -1,5 +1,12 @@
 class AddPublishedStateToScript < ActiveRecord::Migration[5.2]
-  def change
-    add_column :scripts, :published_state, :string
+  def up
+    execute <<-SQL
+        ALTER TABLE scripts ADD published_state enum('in_development', 'pilot', 'beta', 'preview', 'stable');
```

Comment:
it seems like this enum should be documented somewhere other than the migration file. was there a reason not to do the db/structure.sql step? if so, can you find a different way to document this?

Reply:
I can add comments to make this clear. Was there another way of documenting you would think would be helpful? There is also the line in the models that starts with `enum published_state` which helps to point that that field is an enum. I didn't do the db/structure.sql step because they were saying to get rid of db/schema.rb and use db/structure.sql instead and I didn't think that was what we would want to do.

Reply:
> I can add comments to make this clear. Was there another way of documenting you would think would be helpful? There is also the line in the models that starts with enum published_state which helps to point that that field is an enum.

I think the key thing to document is that if we add a new state in the future, updating the shared constants is not enough -- you'll also have to update the database to accept the new value.

> I didn't do the db/structure.sql step because they were saying to get rid of db/schema.rb and use db/structure.sql instead and I didn't think that was what we would want to do.

I see - yes, we probably don't want to make that change, then.

TBH without the ability for the enum to be reflected in schema.rb or the model file, I would vote that disadvantages of the DB enum option seem to outweigh the advantages. Let's go with the simpler option of using a string column for now, and if we decide to move in this direction as a team this should be easy to change later. It'll be really nice that when you update the shared constant, the validation logic will get automatically updated.

Reply:
@Hamms any thoughts before I go back to the string? I'll wait till 5pm PT today to make the change to give you a chance to weigh in.

Reply:
Alright its back to a string now.

Reply:
string seems fine to me, but I'm glad we looked into this!